core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -163,6 +169,12 @@ class HadoopMapReduceCommitProtocol(
   }
 
   override def commitJob(jobContext: JobContext, taskCommits: Seq[TaskCommitMessage]): Unit = {
+    // first delete the should delete special file
+    val committerFs = jobContext.getWorkingDirectory.getFileSystem(jobContext.getConfiguration)
```

Comment:
will this be different from `stagingDir.getFileSystem(jobContext.getConfiguration)`?

Reply:
StagingDir is not always be valid hadoop path, but the JobContext work dir always be.

Reply:
can we change other places in this method to use the `fs` created here?

Reply:
I'm not sure you can guarantee that the working dir is always the dest FS. At least with @rdblue's committers, task attempts work dirs are in file:// & task commit (somehow) gets them to the destFS in a form where job commit will make them visible.

Reply:
I change my code. 
I now record every `FileSystem` will delete the path with a map structure. And Don't assume that they will use the same `FileSystem`.